python-sdk/tutorials/automl-with-azureml/ONNX_batch_model_generator_automl_for_images.py

```diff
@@ -0,0 +1,224 @@
+import onnx
+import torch
+import argparse
+from azureml.core import Experiment
+from azureml.core.workspace import Workspace
+from azureml.train.automl.run import AutoMLRun
+from torchvision.ops._register_onnx_ops import _onnx_opset_version
+from azureml.automl.dnn.vision.common.logging_utils import get_logger
+from azureml.automl.dnn.vision.common.model_export_utils import load_model
+from azureml.automl.dnn.vision.object_detection_yolo.models.common import Conv, Hardswish
+
+logger = get_logger(__name__)
+# constants
+TORCH_MODEL_PATH = 'model.pt'
+
+
+def export_onnx_model(model, dummy_input, input_names, output_names, dynamic_axes, file_path, device):
+    """
+    Export the pytorch model to onnx model file.
+
+    :param model: Model object for faster_rcnn or yolo or mask_rcnn
+    :type model: torchvision.models.detection
+    :param dummy_input: dummy input to be used for ONNX model
+    :type dummy_input: torch tensor
+    :param input_names: Input names to be used for ONNX model
+    :type input_names: dict
+    :param output_names: Output names to be used for ONNX model
+    :type output_names: dict
+    :param dynamic_axes: Dynamic axes to be used for ONNX model
+    :type dynamic_axes: dict
+    :param file_path: File path to save the exported onnx model
+    :type file_path: str
+    :param device: Device where model should be run
+    :type device: str
+    """
+
+    model.eval()
+    model.to(device)
+    torch.onnx.export(model,
+                      dummy_input,
+                      file_path,
+                      opset_version=_onnx_opset_version,
+                      input_names=input_names,
+                      output_names=output_names,
+                      dynamic_axes=dynamic_axes)
+
+
+def generate_onnx_batch_model(batch_size, height_onnx, width_onnx, task_type, best_child_run,
+                              model_settings, model_name, device):
+    """
+    Get the onnx batch model and verify the schema.
+
+    :param batch_size: Batch size needed for batch inference
+    :type batch_size: int
+    :param height_onnx: Height of the image to be used for ONNX model
+    :type height_onnx: int
+    :param width_onnx: Width of the image to be used for ONNX model
+    :type width_onnx: int
+    :param task_type: Type of vision task from image classification, object detection, segmentation
+    :type task_type: str
+    :param best_child_run: Best child run object of an experiment
+    :type best_child_run: azureml.core.run.Run
+    :param model_settings: Model parameters with key, values
+    :type model_settings: dict
+    :param model_type: Model name
+    :type model_type: str
+    :param device: Device on which the model has to be generated
+    :type device: str
+    """
+
+    # download the pytorch model weights
+    best_child_run.download_file(name='train_artifacts/model.pt', output_file_path=TORCH_MODEL_PATH)
```

Comment:
'train_artifacts/model.pt' is defined as a constant in our package, better use that one